ribbon/src/main/java/feign/ribbon/RibbonClient.java

```diff
@@ -57,12 +64,33 @@ public Response execute(Request request, Request.Options options) throws IOExcep
     }
   }
 
-  private LBClient lbClient(String clientName) {
-    IClientConfig config = ClientFactory.getNamedConfig(clientName);
-    ILoadBalancer lb = ClientFactory.getNamedLoadBalancer(clientName);
-    return new LBClient(delegate, lb, config);
+  /**
+   * Returns a cached version of the {@link LBClient} if it exists otherwise a new {@link LBClient}
+   * is created and added to the cache.
+   * <p>
+   * Ribbon does not expect multiple LBClient instances for the same client. This introduces a cache
+   * to avoid redundantly creating them.
+   * 
+   * @param clientName - The clientName that the {@link LBClient} will be cache against
+   * @return
+   */
+  private LBClient getOrCreateLBClient(String clientName) {
+    if (lbClientCache.containsKey(clientName)) {
```

Comment:
If we do got with this rather than the LoadingCache could we make this method protected?Locally I would prefer an override to use a LoadingCache?